.github/workflows/aws-deploy.yml

```diff
@@ -9,6 +9,13 @@ on:
   push:
     branches: ['*']
 
+env:
+  CONTEXT: ${{ github.REF_NAME == 'prod' && 'prod' || 'dev' }}
```

Comment:
This does not seem like a proper bash expression.  have you tested this? 

Also I think all you need here is `CONTEXT: ${{ github.REF_NAME }}` because the condition on the branch is already expressed in `if` condition in the cdk-deploy job.